main.py

```
@@ -1,131 +1,126 @@
+            "views": int(stats.get("viewCount", 0)),
+            "likes": int(stats.get("likeCount", 0)),
+            "comments": int(stats.get("commentCount", 0)),
+            "description": snippet.get("description", "N/A"),
+            "thumbnail_url": snippet["thumbnails"].get("high", {}).get("url", "N/A"),
+            "duration": str(isodate.parse_duration(content_details.get("duration", "PT0S"))),
+            "tags": snippet.get("tags", []),
+            "category_id": snippet.get("categoryId", "N/A"),
+            "channel_id": snippet.get("channelId", "N/A")
         }
+    except googleapiclient.errors.HttpError as e:
+        raise HTTPException(status_code=500, detail=f"YouTube API error: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)